tests/integration/test_local_globals.py

```diff
@@ -0,0 +1,30 @@
+from contextlib import contextmanager
+
+
+@contextmanager
+def restorable_env():
+    import os
+    current_env = os.environ.copy()
+
+    try:
+        yield
+    finally:
+        os.environ.clear()
+        os.environ.update(current_env)
```

Comment:
All tests will share the same environment, right? This could cause issues for tests running concurrently that also inspect the process environment, especially clearing the environment. To minimize the risk of other tests missing some information from the environment, what do you think of:
```suggestion
        for k, v in os.environ.enumerate():
            if k not in current_env:
                del os.environ[k]
            else if v != current_env[k]:
                os.environ[k] = current_env[k]
```